cpp/src/gandiva/precompiled/extended_math_ops.cc

```diff
@@ -367,4 +367,33 @@ gdv_float64 get_scale_multiplier(gdv_int32 scale) {
   return power_float64_float64(10.0, scale);
 }
 
+// convert input unsigned long to its binary representation
+void bin(uint64_t n, char* ret, int32_t* position) {
```

Comment:
Mark it as force inline. Also the recursion won't be inlined. Can you replace the logic with iteration.